internal/adapters/operation/operation_storage_redis_test.go

```diff
@@ -876,3 +876,140 @@ func TestCleanOperationsHistory(t *testing.T) {
+						Event:     "some-event",
+					},
+				},
+			},
+			{
+				ID:             "some-op-id-4",
+				SchedulerName:  schedulerName,
+				Status:         operation.StatusFinished,
+				DefinitionName: "test-definition",
+				CreatedAt:      now.Add(-29 * time.Hour),
+				Input:          []byte("hello test"),
+				ExecutionHistory: []operation.OperationEvent{
+					{
+						CreatedAt: time.Date(1999, time.November, 19, 6, 12, 15, 0, time.UTC),
+						Event:     "some-event",
+					},
+				},
+			},
+		}
+
+		t.Run("if operations history is not empty it deletes all expired operations", func(t *testing.T) {
+			client := test.GetRedisConnection(t, redisAddress)
+			clock := clockmock.NewFakeClock(time.Now())
+			operationsTTlMap := map[Definition]time.Duration{}
+			storage := NewRedisOperationStorage(client, clock, operationsTTlMap)
+			expectedOperations := []*operation.Operation{operations[1], operations[0]}
+
+			for _, op := range operations {
+				executionHistoryJson, err := json.Marshal(op.ExecutionHistory)
+				require.NoError(t, err)
+
+				err = client.ZAdd(context.Background(), storage.buildSchedulerHistoryOperationsKey(op.SchedulerName), &redis.Z{
+					Member: op.ID,
+					Score:  float64(op.CreatedAt.Unix()),
+				}).Err()
+				require.NoError(t, err)
+
+				err = client.HSet(context.Background(), storage.buildSchedulerOperationKey(op.SchedulerName, op.ID), map[string]interface{}{
+					idRedisKey:                 op.ID,
+					schedulerNameRedisKey:      op.SchedulerName,
+					statusRedisKey:             strconv.Itoa(int(op.Status)),
+					definitionNameRedisKey:     op.DefinitionName,
+					createdAtRedisKey:          op.CreatedAt.Format(time.RFC3339Nano),
+					definitionContentsRedisKey: op.Input,
+					executionHistoryRedisKey:   executionHistoryJson,
+				}).Err()
+				require.NoError(t, err)
+			}
+
+			operationsReturned, err := storage.ListSchedulerFinishedOperations(context.Background(), schedulerName)
+			assert.NoError(t, err)
+			assert.NotEmptyf(t, operationsReturned, "expected at least one operation")
+			assert.Equal(t, expectedOperations, operationsReturned)
+
+			err = storage.CleanExpiredOperations(context.Background(), schedulerName)
+			assert.NoError(t, err)
+
+			operationIds, err := client.ZRangeByScore(context.Background(), fmt.Sprintf("operations:%s:lists:history", schedulerName), &redis.ZRangeBy{
+				Min: "-inf",
+				Max: "+inf",
+			}).Result()
+			require.NoError(t, err)
+			assert.NotEmpty(t, operationIds)
+
+			for _, op := range expectedOperations {
+				result, _ := client.HGetAll(context.Background(), fmt.Sprintf("operations:%s:%s", schedulerName, op.ID)).Result()
+				require.Equal(t, op.ID, result[idRedisKey])
+				require.Equal(t, op.SchedulerName, result[schedulerNameRedisKey])
+				require.Equal(t, op.DefinitionName, result[definitionNameRedisKey])
+				require.EqualValues(t, op.Input, result[definitionContentsRedisKey])
+			}
```

Comment:
Here we are asserting that all expected operations (operations[0] and operations[1]) exists in the DB as hashMaps, this test will always pass.

The assertion we should do here is that the operations history sorted set does not contain the expired operations IDs anymore, something like this:
```
			operationsIDs, err := client.ZRangeByScore(ctx, r.buildSchedulerHistoryOperationsKey(schedulerName), &redis.ZRangeBy{
				Min: "-inf",
				Max: "+inf",
			}).Result()
			assert.NoError(t, err)
			assert.NotContains(t, operationsIDs, operations[2].ID, operations[3].ID)
			assert.Contains(t, operationsIDs, operations[0].ID, operations[1].ID)
```
Here, we are considering that operations[0] and operations[1] are not expired, so they still exists in the history, but operations[2] and operations[3] were expired, so our clean up method removed them from the history